syntax = "proto3";

package App.Protobuf;

option php_namespace = "App\\Protobuf";
option php_metadata_namespace = "App\\Protobuf\\GPBMetadata";

message ApiToSoundExtractor {
    MediaPod mediaPod = 1;
}

message SoundExtractorToApi {
    MediaPod mediaPod = 1;
}

message ApiToSubtitleGenerator {
    MediaPod mediaPod = 1;
}

message SubtitleGeneratorToApi {
    MediaPod mediaPod = 1;
}

message ApiToSubtitleMerger {
    MediaPod mediaPod = 1;
}

message SubtitleMergerToApi {
    MediaPod mediaPod = 1;
}

message ApiToSubtitleTransformer {
    MediaPod mediaPod = 1;
}

message SubtitleTransformerToApi {
    MediaPod mediaPod = 1;
}

message ApiToVideoFormatter {
    MediaPod mediaPod = 1;
}

message VideoFormatterToApi {
    MediaPod mediaPod = 1;
}

message ApiToSubtitleIncrustator {
    MediaPod mediaPod = 1;
}

message SubtitleIncrustatorToApi {
    MediaPod mediaPod = 1;
}

enum MediaPodStatus {
    UPLOAD_COMPLETE = 0;

    SOUND_EXTRACTOR_PENDING = 1;
    SOUND_EXTRACTOR_COMPLETE = 2;
    SOUND_EXTRACTOR_ERROR = 3;

    SUBTITLE_GENERATOR_PENDING = 4;
    SUBTITLE_GENERATOR_COMPLETE = 5;
    SUBTITLE_GENERATOR_ERROR = 6;

    SUBTITLE_MERGER_PENDING = 7;
    SUBTITLE_MERGER_COMPLETE = 8;
    SUBTITLE_MERGER_ERROR = 9;

    SUBTITLE_TRANSFORMER_PENDING = 10;
    SUBTITLE_TRANSFORMER_COMPLETE = 11;
    SUBTITLE_TRANSFORMER_ERROR = 12;

    SUBTITLE_INCRUSTATOR_PENDING = 13;
    SUBTITLE_INCRUSTATOR_COMPLETE = 14;
    SUBTITLE_INCRUSTATOR_ERROR = 15;

    VIDEO_FORMATTER_PENDING = 16;
    VIDEO_FORMATTER_COMPLETE = 17;
    VIDEO_FORMATTER_ERROR = 18;

    RESIZING = 19;
    RESIZED = 20;

    READY_FOR_EXPORT = 21;
    ERROR = 22;
}

enum PresetSubtitleShadow {
    SHADOW_NONE = 0;
    SHADOW_SOFT = 1;
    SHADOW_MEDIUM = 2;
    SHADOW_HARD = 4;
}

enum PresetSubtitleOutlineThickness{
    OUTLINE_NONE = 0;
    OUTLINE_SOFT = 1;
    OUTLINE_MEDIUM = 2;
    OUTLINE_HARD = 4;
}

enum PresetSubtitleFont {
    ARIAL = 0;
    TIMES_NEW_ROMAN = 1;
    COURIER_NEW = 2;
}

enum FormatStyle {
    DEFAULT = 0;
    ZOOMED_916 = 1;
}

message MediaPod {
    string uuid = 1;
    string userUuid = 2;
    Video originalVideo = 3;
    Video video = 4;
    string status = 5;
    Preset preset = 6;
    Video processedVideo = 7;
}

message Video {
    string name = 1;
    string mimeType = 2;
    int64 size = 3;
    int64 length = 4;
    string subtitle = 5;
    string ass = 6;
    repeated string subtitles = 7;
    repeated string audios = 8;
    optional Format format = 9;
}

message Preset {
    string subtitleFont = 1;
    string subtitleSize = 2;
    string subtitleColor = 3;
    string subtitleBold = 4;
    string subtitleItalic = 5;
    string subtitleUnderline = 6;
    string subtitleOutlineColor = 7;
    string subtitleOutlineThickness = 8;
    string subtitleShadow = 9;
    string subtitleShadowColor = 10;
}

message Format {
    string style = 1;
}
